controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -432,4 +449,25 @@ void dynamic_channel_selection_task::fsm_move_state(eState new_state)
                     << s_ar_states[int(new_state)];
     m_fsm_state = new_state;
 }
-bool dynamic_channel_selection_task::fsm_in_state(eState state) { return m_fsm_state == state; }
\ No newline at end of file
+bool dynamic_channel_selection_task::fsm_in_state(eState state) { return m_fsm_state == state; }
+
+bool dynamic_channel_selection_task::start_scan()
+{
+    if (m_scanning_task_id != -1) {
+        return false;
+    }
+    // set scanning task id to current task id
+    m_scanning_task_id = id;
+    return true;
+}
+
+bool dynamic_channel_selection_task::finish_scan()
+{
+    if (m_scanning_task_id != id) {
```

Comment:
Since now you're also calling this method from the destructor, you should also check that `m_scanning_task_id != -1`, otherwise it will always be printed:
```suggestion
    if (m_scanning_task_id != -1 && m_scanning_task_id != id) {
```

Reply:
I will add a separate check:
```
if (m_scanning_task_id == -1) {
    return true;
}
```